### ISSUE TITLE
Which county changed the most?

### ISSUE THREAD

**Issue by arilamstein** [arilamstein/censusdis-streamlit#11]
It's nice that the app now lets you select a (state, county, demographic) tuple and see a time series graph of the value and a bar chart of the year-over-year percent change of that value.

But:
  1. This app is focused on covid, so the change from 2019-2021 is most important (reminder: 2020 data does not exist)
  2. It seems natural to ask the question: "Well, ok, this is how *my* county changed. But which county changed the most for this demographic?"

I think the app should continue to help people understand how their particular county changed. But it should also facilitate this exploration of "which regions changed the most from 2019-2021?" 

This can probably be done with a combination of a windowing operation and a groupby on the main dataframe in the app. See https://pandas.pydata.org/pandas-docs/stable/user_guide/window.html

**Comment by arilamstein** [comment]
So, here's code that appears to do what I want. The results are interesting, in that the county where I live (San Francisco) appears to have had the 3rd largest drop in total population during Covid, at -7.5%. The 3 counties with the largest gain were all in Texas, but quite small (so it's easier to get a large % increase). Ideally the app should let people toggle between raw numbers and percentage changes, as you kinda need both to get a feel for the data.

```
import pandas as pd

df = pd.read_csv('county_data.csv')

# Remove columns we don't need
df = df.drop(columns = ['B19013_001E', 'B25058_001E', 'STATE_NAME', 'COUNTY_NAME'])

# Do the groupby -> percent change. Verify that when you hit 2005 for the next county, the data is NaN again
df['B01001_001E_PERCENT_CHANGE'] = df.groupby(['STATE', 'COUNTY'])['B01001_001E'].pct_change() * 100
df.head(20) 

# Filter to 2021 and look at min / max values 
(
    df
    .loc[df['YEAR'] == 2021]
    .dropna(subset='B01001_001E_PERCENT_CHANGE')
    .sort_values('B01001_001E_PERCENT_CHANGE')
)
```
The `NaN` values I drop in the last line are all counties with a population slightly over 65k. As a reminder, the 1-year ACS only reports data on counties with a population >= 65k. So each year there are probably a handful of counties that will pop in and out of the data. That is, I dn't think it's anything to worry about.

**Comment by arilamstein** [comment]
I just added a table to the bottom of the screen showing the rankings in dev. The representation is a dataframe with first column "Rank" and last column "% Change".

There is an issue where the page is cluttered, and confusing because the two graphs are a single county over all time, and the df is a comparison/ranking of counties between 2019-2021. 

I should look into other ways to convey this info so the user is a bit clearer on what they're seeing. The first thing that comes to mind is tabs: https://docs.streamlit.io/develop/api-reference/layout/st.tabs

**Comment by arilamstein** [comment]
Closed by #27
